src/Core/Generator/SchemaGenerator.cs

```diff
@@ -44,7 +44,7 @@ private SchemaGenerator(List<JsonDocument> data, string containerName, RuntimeCo
                 // Populate entity and singular name mapping if configuration is provided.
                 foreach (KeyValuePair<string, Entity> item in config.Entities)
                 {
-                    _entityAndSingularNameMapping.Add(item.Value.GraphQL.Singular, item.Key);
+                    _entityAndSingularNameMapping.Add(item.Value.GraphQL.Singular.Pascalize(), item.Key);
                 }
```

Comment:
if you only store the key in pascal format, would non-pascal singular name still work?

Reply:
yes..in existing test, it is in lowercase and still works.